File: LC860.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        f = t = 0

        for i in bills:
            if i == 5:
                f += 1
            elif i == 10:
                if f == 0:
                    return False
                f -= 1
                t += 1
            else:
                if f > 0 and t > 0:
                    t -= 1
                    f -= 1
                elif f >= 3:
                    f -= 3
                else:
                    return False
        
        return True

File: LC860.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        
        int five = 0, ten = 0;
        
        for(int i : bills){
            if(i == 5)
                five++;
            else if(i == 10){
                if(five == 0)
                    return false;
                five--;
                ten++;
            }
            else{
                if(five > 0 and ten > 0){
                    ten--;
                    five--;
                }
                else if(five >= 3)
                    five -= 3;
                else
                    return false;
            }
        }
        return true;
    }
};
